gix/src/repository/blame.rs

```diff
@@ -0,0 +1,43 @@
+#[cfg(feature = "blame")]
+use gix_hash::ObjectId;
+#[cfg(feature = "blame")]
+use gix_ref::bstr::BStr;
+
+use crate::{
+    repository::{commit_graph_if_enabled, diff_resource_cache},
+    Repository,
+};
+
+/// The error returned by [`Repository::blame_file()`].
+#[derive(Debug, thiserror::Error)]
+#[allow(missing_docs)]
+pub enum Error {
+    #[error(transparent)]
+    CommitGraphIfEnabled(#[from] commit_graph_if_enabled::Error),
+    #[error(transparent)]
+    DiffResourceCache(#[from] diff_resource_cache::Error),
+    #[error(transparent)]
+    Blame(#[from] gix_blame::Error),
+}
```

Comment:
In the absence of custom 'connected' blame related types, I think this whole module should be in `gix::repository::blame`, with the error type 'where the other errors types are' - there is pattern for this.

Alternatively `gix_blame::Outcome` would have to be turned into a duplicate structure with a reference to `repo: &Repository`, which doesn't seem worth it - it's the bane of the `gix` crate as it always has to repeat whole types just to add some convenience.

Reply:
I’ve moved the enum, following the example of `diff_tree_to_tree::Error` and others. Was that correct?